src/author/views.py

```diff
@@ -335,6 +349,126 @@ def revision(request, revision_id, submission_id):
+    form_class = core_forms.CustomEmailForm
+
+    @method_decorator(login_required)
+    def dispatch(self, request, *args, **kwargs):
+        self.submission = get_object_or_404(
+            models.Book,
+            pk=self.kwargs['submission_id']
+        )
+        self.revision = get_object_or_404(
+            revision_models.Revision,
+            pk=self.kwargs['revision_id']
+        )
+
+        return super(RevisionCompletionEmail, self).dispatch(
+            request,
+            *args,
+            **kwargs
+        )
+
+    def get_form_kwargs(self):
+        """Renders the email body and subject for editing using the form
+        """
+        kwargs = super(RevisionCompletionEmail, self).get_form_kwargs()
+
+        if self.revision.requestor:
+            recipient_greeting = email.get_email_greeting(
+                recipients=[self.revision.requestor]
+            )
+        else:
+            recipient_greeting = 'Dear sir or madam'
+
+        email_context = {
+            'greeting': recipient_greeting,
+            'submission': self.submission,
+            'sender': self.request.user,
+        }
+
+        email_body = email.get_email_content(
+            request=self.request,
+            setting_name='author_revisions_completed',
+            context=email_context,
+        )
+        email_subject = 'Review completed for {title}: {subtitle}'.format(
+            title=self.submission.title,
+            subtitle=self.submission.subtitle,
+        )
+
+        kwargs['initial'] = {
+            'email_subject': email_subject,
+            'email_body': email_body,
+        }
+
+        return kwargs
+
+    def get_context_data(self, **kwargs):
+        context = super(RevisionCompletionEmail, self).get_context_data(**kwargs)
+        context['submission'] = get_object_or_404(
+            models.Book,
+            pk=self.kwargs['submission_id']
+        )
+        return context
+
+    def form_valid(self, form):
+        attachments = handle_multiple_email_files(
+            request_files=self.request.FILES.getlist('attachments'),
+            file_owner=self.request.user,
+        )
+
+        other_editors = []
+        for book_editor in self.submission.book_editors.all():
+            if book_editor != self.revision.requestor:
+                other_editors.append(book_editor)
+
+        series_editor = self.submission.get_series_editor()
+        if series_editor and series_editor != self.revision.requestor:
+            other_editors.append(series_editor)
+
+        copy_email_addresses = []
+        if self.revision.requestor:
+            recipient_email_addresses = [self.revision.requestor.email]
+            copy_email_addresses.extend(
+                [editor.email for editor in other_editors]
+            )
+        else:
+            recipient_email_addresses = [
+                editor.email for editor in other_editors
+            ]
+
+        # If no other editors are found, use press editors
```

Comment:
Will this be sent to the editor of every press? Should it not be specific to the press that the revision was done for?

Reply:
It would be sent to the users with the 'press_editor' role on the Rua instance for a particular press. 

As I'm 99.9% sure there would always be a requestor for the revision, I don't think this is necessary and may remove. If recipient_email_addresses ended up empty, it would probably indicate something wrong elsewhere in the app.

Reply:
Ah right, I forgot a separate instance of Rua is run for every press.